File: frontend/src/api/customers.ts
import { request } from "./client";
import type { Customer, CustomerInput } from "../types/customer";

export const listCustomers = () => request<Customer[]>("/customers");
export const getCustomer = (id: number) => request<Customer>(`/customers/${id}`);
export const createCustomer = (payload: CustomerInput) =>
  request<Customer>("/customers", { method: "POST", body: JSON.stringify(payload) });

File: backend/app/api/routes/customers.py
from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.schemas.customer import CustomerCreate, CustomerRead
from app.services import customers

router = APIRouter(prefix="/customers", tags=["customers"])


@router.post("", response_model=CustomerRead, status_code=status.HTTP_201_CREATED)
def create_customer(payload: CustomerCreate, db: Session = Depends(get_db)) -> CustomerRead:
    return customers.create_customer(db, payload)


@router.get("", response_model=list[CustomerRead])
def list_customers(db: Session = Depends(get_db)) -> list[CustomerRead]:
    return customers.list_customers(db)


@router.get("/{customer_id}", response_model=CustomerRead)
def get_customer(customer_id: int, db: Session = Depends(get_db)) -> CustomerRead:
    return customers.get_customer(db, customer_id)
